File: Cpp/2023/Day01/day01_v1.cpp
#include <iostream>
#include <string>
#include <vector>

#include "../../../../tools/DataFromTxt.hpp"

using namespace std;

// Day 1 - Part 1 of Advent of Code 2023

int main()
{
    //string path = "D:/.Documents/AdventOfCode/Data/2023/day01_test1.txt";
    string path = "D:/.Documents/AdventOfCode/Data/2023/day01.txt";
    DataFromTxt data(path);
    vector<string> lines = data.readLines();

    string num("1234567890");

    int sum = 0;
    for (string line : lines)
    {
        size_t first_num = line.find_first_of(num);
        size_t last_num = line.find_last_of(num);
        sum += stoi(string {line[first_num],line[last_num]});
    }

    cout << "Result : " << sum << endl;
    // Result : 53194
}

File: Python/2023/day01_v1.py
# Day 1, part 1 of Advent of Code 2023

#data_file = open('../../Data/2023/day01_test1.txt', 'r')

data_file = open('../../Data/2023/day01.txt', 'r')
data_lines = []
for line in data_file:
    data_lines.append(line)
data_file.close()

cifers = ['0','1','2','3','4','5','6','7','8','9']
sum = 0

for line in data_lines :
    start = 0
    end = len(line)-1
    while start < len(line) :
        if line[start] in cifers :
            break
        else :
            start += 1
    while end >= 0 :
        if line[end] in cifers :
            break
        else :
            end -= 1

    if start <= end :
        line = int(line[start]+line[end])
        sum += line
    else :
        print('Error at line : ', line)

print('Sum = ', sum)
# Answer is 53194